### PR TITLE
Depth-first search

### PR DESCRIPTION
**Pull request by abircb**
Added Depth-first search to the [Searches](https://github.com/TheAlgorithms/Java/tree/master/Searches) repository and applied the suggestions from code review by @yanglbme

### CODE REVIEW

**Comment by yanglbme** [comment]
Hi @abircb , the class `BinaryTree` has already been defined in package `dataStructures`.

image

**Comment by abircb** [comment]
Is there a problem if I define my own Binary Tree class in the DepthFirstSearch method?

**Comment by yanglbme** [comment]
@abircb yeah, the `DepthFirstSearchTest` cannot find out the `BinaryTree` class you defined.

**Comment by abircb** [comment]
@yanglbme I have fixed that and a few other problems- removed the deprecated Assert.assertEquals()

**Comment by abircb** [comment]
Thanks for the code review @yanglbme

File: src/test/java/com/search/DepthFirstSearchTest.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package src.test.java.com.search;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import src.main.java.com.search.DepthFirstSearch;
+import src.main.java.com.search.BinaryTree;
```
Comment (yanglbme) [inline src/test/java/com/search/DepthFirstSearchTest.java]: 'src.main.java.com.search.BinaryTree' is not public in 'src.main.java.com.search'. Cannot be accessed from outside package.

Comment (abircb) [reply src/test/java/com/search/DepthFirstSearchTest.java]: any on how I can fix this?

**Comment by ayaankhan98** [comment]
Hey please make PR to master branch.

**Pull request closed without merging**